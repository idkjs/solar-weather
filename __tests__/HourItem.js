import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import HourItem from '../lib/js/app/components/hourItem';

test('renders correctly', () => {
  const tree = renderer.create(<HourItem
    unit="c"
    timeType="24"
    temperature={0.0}
    icon="sunny"
    timezone="America/New_York"
    time={1515069527}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
