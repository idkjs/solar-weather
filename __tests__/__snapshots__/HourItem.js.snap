// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityLiveRegion={undefined}
  accessibilityTraits={undefined}
  accessibilityViewIsModal={undefined}
  accessible={undefined}
  collapsable={undefined}
  hitSlop={undefined}
  importantForAccessibility={undefined}
  needsOffscreenAlphaCompositing={undefined}
  onAccessibilityTap={undefined}
  onLayout={undefined}
  onMagicTap={undefined}
  pointerEvents={undefined}
  removeClippedSubviews={undefined}
  renderToHardwareTextureAndroid={undefined}
  shouldRasterizeIOS={undefined}
  style={
    Object {
      "alignItems": "center",
      "height": "100.00%",
      "paddingLeft": 15,
      "paddingRight": 15,
    }
  }
  testID={undefined}
>
  <Text
    accessible={true}
    adjustsFontSizeToFit={undefined}
    allowFontScaling={true}
    ellipsizeMode="tail"
    minimumFontScale={undefined}
    numberOfLines={undefined}
    onLayout={undefined}
    onLongPress={undefined}
    onPress={undefined}
    pressRetentionOffset={undefined}
    selectable={undefined}
    selectionColor={undefined}
    style={
      Object {
        "color": "#EFEFEF",
        "fontWeight": "600",
        "marginTop": 15,
        "textAlign": "center",
      }
    }
    suppressHighlighting={undefined}
    testID={undefined}
    textBreakStrategy={undefined}
  >
    07:38
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityLiveRegion={undefined}
    accessibilityTraits={undefined}
    accessibilityViewIsModal={undefined}
    accessible={undefined}
    collapsable={undefined}
    hitSlop={undefined}
    importantForAccessibility={undefined}
    needsOffscreenAlphaCompositing={undefined}
    onAccessibilityTap={undefined}
    onLayout={undefined}
    onMagicTap={undefined}
    pointerEvents={undefined}
    removeClippedSubviews={undefined}
    renderToHardwareTextureAndroid={undefined}
    shouldRasterizeIOS={undefined}
    style={
      Object {
        "alignItems": "center",
        "height": "50.00%",
        "justifyContent": "center",
        "width": "50.00%",
      }
    }
    testID={undefined}
  >
    <Image
      accessibilityLabel={undefined}
      accessible={undefined}
      blurRadius={undefined}
      capInsets={undefined}
      defaultSource={undefined}
      onError={undefined}
      onLayout={undefined}
      onLoad={undefined}
      onLoadEnd={undefined}
      onLoadStart={undefined}
      onPartialLoad={undefined}
      onProgress={undefined}
      resizeMethod={undefined}
      resizeMode={undefined}
      source={
        Object {
          "testUri": "../../../assets/weather_icons/sunny.png",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "marginTop": -5,
          "resizeMode": "contain",
          "width": "170.00%",
        }
      }
      testID={undefined}
    />
  </View>
  <Text
    accessible={true}
    adjustsFontSizeToFit={undefined}
    allowFontScaling={true}
    ellipsizeMode="tail"
    minimumFontScale={undefined}
    numberOfLines={undefined}
    onLayout={undefined}
    onLongPress={undefined}
    onPress={undefined}
    pressRetentionOffset={undefined}
    selectable={undefined}
    selectionColor={undefined}
    style={
      Object {
        "color": "#EFEFEF",
        "fontSize": 14,
      }
    }
    suppressHighlighting={undefined}
    testID={undefined}
    textBreakStrategy={undefined}
  >
    0Â°
  </Text>
</View>
`;
